plaidml/edsl/edsl_test.cc

```diff
@@ -46,6 +46,44 @@ Tensor Softmax(const Tensor& X) {
   return E / N;
 }
 
+TEST(CppEdsl, LowerPrecision) {
+  auto A = Placeholder(DType::FLOAT64, {3, 3});
+  auto C = A + 1 + 2.0;
+
+  Program program("lower_precision", {C}, DType::FLOAT32, DType::INT32);
+
+  // consatnt_types pass doesn't lower precision to float32/int32 until program is compiled
+  EXPECT_THAT(program, Eq(R"#(
+
+!i32 = type tensor<!eltwise.i32>
+!f32 = type tensor<!eltwise.f32>
+module {
+  func @lower_precision(%arg0: tensor<3x3x!eltwise.f64>) -> tensor<3x3x!eltwise.f64> {
+    %c1 = "eltwise.sconst"() {value = 1 : i64} : () -> !i32
+    %cst = "eltwise.sconst"() {value = 2.000000e+00 : f64} : () -> !f32
+    %0 = "eltwise.add"(%arg0, %c1) : (tensor<3x3x!eltwise.f64>, !i32) -> tensor<3x3x!eltwise.f64>
+    %1 = "eltwise.add"(%0, %cst) : (tensor<3x3x!eltwise.f64>, !f32) -> tensor<3x3x!eltwise.f64>
+    return %1 : tensor<3x3x!eltwise.f64>
```

Comment:
This seems like `floatx` isn't working properly here, wouldn't we expect to see a `f32` return type?

Reply:
correct; fixed